File: 3430-count-days-without-meetings/count-days-without-meetings.cpp
class Solution {
public:
    int countDays(int days, vector<vector<int>>& meetings) {
        sort(meetings.begin(), meetings.end());
        int n = meetings.size();
        int end = INT_MIN;
        int ans = (meetings[0][0] - 1);
        for(int i=0; i<meetings.size()-1; i++){
            end = max(end, meetings[i][1]);
            if(meetings[i+1][0] > end){
                ans += (meetings[i+1][0] - end - 1);
                end = meetings[i+1][1];
            }
        }
        ans += days - max(end, meetings[n-1][1]);
        return ans;
    }
};

File: 3430-count-days-without-meetings/count-days-without-meetings.py
class Solution:
    def countDays(self, days: int, meetings: List[List[int]]) -> int:
        
        if len(meetings) == 0:
            return days
        
        meetings.sort(key=lambda x: [x[0],-x[1]])

        s = meetings[0][0]
        e = meetings[0][1]

        count = s-1
        idx = 0
        while(idx < len(meetings)):
            if meetings[idx][0] <= e:
                e = max(e, meetings[idx][1])
            else:
                count += meetings[idx][0]-e-1
                s = meetings[idx][0]
                e = meetings[idx][1]
            idx += 1
        count += (days-e)

        return count
        